packages/inferno-router/__tests__/router.spec.jsx

```diff
@@ -262,6 +273,66 @@ describe('Router (jsx)', () => {
 				() => render(<RouterContext location={ null }/>, container)
 			).to.throw(TypeError);
 		});
+		it('should fail when `matched` is not provided', () => {
+			expect(
+				() => render(<RouterContext location={ '/' } matched={ null }/>, container)
+			).to.throw(TypeError);
+		});
+		it('should pass when `location` is provided', () => {
+			const url = '/';
+			const matched = <GoodComponent/>;
+			const actual = render(<RouterContext location={ url } matched={ matched }/>, container);
+			expect(actual.props.location).to.equal(url);
+			expect(actual.props.matched).to.equal(matched);
+		});
+		it('should pass when `location` is provided and has percent encoded value', () => {
+			const url = '/100%25';
+			const matched = <GoodComponent/>;
+			const actual = render(<RouterContext location={ url } matched={ matched }/>, container);
+			expect(actual.props.location).to.equal(url);
+			expect(actual.props.matched).to.equal(matched);
+		});
+	});
+	describe('#match', () => {
+		it('should find route when url has normal value', () => {
+			const url = '/search/foo';
+			const history = createMemoryHistory();
+			history.push(url);
+
+			const router = <Router history={ history }>
+				<Route path="/" component={ BadComponent }>
+					<Route path="search/:searchData" component={ GoodComponent }/>
+				</Route>
+			</Router>;
+			const renderProps = match(router, url);
+
+			const actual = render(<RouterContext {...renderProps}/>, container);
+
+			expect(actual.props.location).to.equal('/');
+			expect(actual.props.matched.props.history.location.pathname).to.equal(url);
+			expect(actual.props.matched.props.params.searchData).to.equal('foo');
+			expect(actual.props.matched.props.children.props.component).to.equal(BadComponent);
+			expect(actual.props.matched.props.children.props.children.props.component).to.equal(GoodComponent);
+		});
+		it('should find route when url has percent encoded value', () => {
+			const url = '/search/100%25';
+			const history = createMemoryHistory();
+			history.push(url);
+			const router = <Router history={ history }>
+				<Route path="/" component={ BadComponent }>
+					<Route path="search/:searchData" component={ GoodComponent }/>
+				</Route>
+			</Router>;
+			const renderProps = match(router, history.location.pathname);
```

Comment:
`history` has the decoded URL. I originally had `match(router, url)` to replicate the behavior of passing in `req.originalUrl` in node, but was getting the same error anyway and thought maybe that was the cause of the problem.